pytorch_toolkit/formula_recognition/im2latex/data/utils.py

```diff
@@ -0,0 +1,510 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import datetime
+import pickle as pkl
+import os
+
+import cv2 as cv
+import numpy as np
+import torch
+from torchvision.transforms import ToTensor, Compose
+import operator
+
+
+from .vocab import END_TOKEN, PAD_TOKEN, START_TOKEN, UNK_TOKEN, Vocab
+import torch.nn.functional as F
+
+COLOR_WHITE = (255, 255, 255)
+
+
+class BatchResizePadToTGTShape():
```

Comment:
I would have 3 separate transforms: Resize, Crop, Pad

Reply:
There are separate transformations for Resize, Crop and Padding. The idea was to make a single transformation to keep original ratio and, using a padding, make a picture have a constant shape.